fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/DimensionFieldMapper.java

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.dimension;
+
+import java.util.Locale;
+
+/**
+ * Maps a Dimension Field value to a string.
+ */
+@FunctionalInterface
+public interface DimensionFieldMapper {
+
+    /**
+     * A converter function for the dimension field Mapping.
+     *
+     * @param dimension The dimension object used to configure the Dimension
+     * @param dimensionField The dimension field object
+     *
+     * @return the converted string for the dimension field
+     */
+    String converter(Dimension dimension, DimensionField dimensionField);
+
```

Comment:
Alternatively, we could also have this interface extend `BiFunction<Dimension, DimensionField, String>`, and this method would just be called `apply`.

We could go _even farther_ and just do away with this interface entirely, and instead have `AvroDimensionRowParser` just take a `BiFunction<Dimension, DimensionField, String>` (its purpose can be made clear enough via a good variable name). One less wrapper concept in the codebase. Though I don't know enough about how we plan to integrate this into the rest of the codebase to know if that's a good idea.